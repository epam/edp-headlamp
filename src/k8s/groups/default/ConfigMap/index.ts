import { K8s } from '@kinvolk/headlamp-plugin/lib';
import { streamResults } from '../../../common/streamResults';
import { ConfigMapKubeObjectConfig } from './config';
import { StreamListProps } from './types';

const {
  name: { pluralForm },
  version,
} = ConfigMapKubeObjectConfig;

export class ConfigMapKubeObject extends K8s.configMap.default {
  static streamList({ namespace, dataHandler, errorHandler }: StreamListProps): () => void {
    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}`;
    return streamResults(url, dataHandler, errorHandler);
  }
}
