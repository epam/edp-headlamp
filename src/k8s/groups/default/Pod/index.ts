import { K8s } from '@kinvolk/headlamp-plugin/lib';

export class PodKubeObject extends K8s.pod.default {}
