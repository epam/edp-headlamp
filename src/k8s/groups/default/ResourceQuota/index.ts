import { K8s } from '@kinvolk/headlamp-plugin/lib';
import { streamResults } from '../../../common/streamResults';
import { ResourceQuotaKubeObjectConfig } from './config';
import { RESOURCE_QUOTA_LABEL_TENANT } from './labels';
import { StreamListProps } from './types';

const {
  name: { pluralForm },
  version,
} = ResourceQuotaKubeObjectConfig;

export class ResourceQuotaKubeObject extends K8s.ResourceClasses.ResourceQuota {
  static streamList({
    namespace,
    tenantNamespace,
    dataHandler,
    errorHandler,
  }: StreamListProps): () => void {

    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}`;
    return streamResults(url, dataHandler, errorHandler, {
      labelSelector: `${RESOURCE_QUOTA_LABEL_TENANT}=edp-workload-${tenantNamespace}`,
    });
  }
}
