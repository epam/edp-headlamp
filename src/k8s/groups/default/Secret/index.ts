import { ApiProxy, K8s } from '@kinvolk/headlamp-plugin/lib';
import { STATUS_COLOR } from '../../../../constants/colors';
import { ICONS } from '../../../../icons/iconify-icons-mapping';
import { KubeObjectListInterface } from '../../../../types/k8s';
import { SecretKubeObjectConfig } from './config';
import { SECRET_LABEL_SECRET_TYPE } from './labels';
import { SecretKubeObjectInterface } from './types';

const {
  name: { pluralForm },
  version,
} = SecretKubeObjectConfig;

export class SecretKubeObject extends K8s.secret.default {
  static getClusterSecretList(
    namespace: string
  ): Promise<KubeObjectListInterface<SecretKubeObjectInterface>> {
    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}?labelSelector=${SECRET_LABEL_SECRET_TYPE}=cluster`;

    return ApiProxy.request(url);
  }

  static getSecretByName(namespace: string, name: string): Promise<SecretKubeObjectInterface> {
    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}/${name}`;

    return ApiProxy.request(url);
  }

  static getStatusIcon(connected: string): [string, string, boolean?] {
    if (connected === undefined) {
      return [ICONS.UNKNOWN, STATUS_COLOR.UNKNOWN];
    }

    if (connected === 'true') {
      return [ICONS.CHECK_CIRCLE, STATUS_COLOR.SUCCESS];
    }
    return [ICONS.CROSS_CIRCLE, STATUS_COLOR.ERROR];
  }
}
