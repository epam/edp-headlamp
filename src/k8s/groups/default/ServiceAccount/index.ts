import { K8s } from '@kinvolk/headlamp-plugin/lib';

export class ServiceAccountKubeObject extends K8s.serviceAccount.default {}
