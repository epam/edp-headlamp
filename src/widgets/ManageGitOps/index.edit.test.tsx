/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../mocks/wrappers/default';
import { ManageGitOps } from './index';

test('renders ManageGitOps Edit component', () => {
  render(
    <TestWrapper>
      <ManageGitOps
        formData={{
          currentElement: {
            apiVersion: 'v2.edp.epam.com/v1',
            kind: 'Codebase',
            // @ts-ignore
            metadata: {
              labels: {
                'app.edp.epam.com/codebaseType': 'system',
              },
              name: 'krci-gitops',
              namespace: 'edp-delivery-vp-dev',
            },
            // @ts-ignore
            spec: {
              buildTool: 'helm',
              ciTool: 'tekton',
              defaultBranch: 'main',
              deploymentScript: 'helm-chart',
              description: 'Custom values for deploy applications',
              emptyProject: false,
              framework: 'gitops',
              gitServer: 'gerrit',
              gitUrlPath: '/krci-gitops',
              jiraIssueMetadataPayload: null,
              lang: 'helm',
              strategy: 'create',
              ticketNamePattern: null,
              type: 'system',
              versioning: {
                startFrom: '0.1.0-SNAPSHOT',
                type: 'edp',
              },
            },
          },
          isReadOnly: true,
          handleDeleteRow: jest.fn(),
          permissions: {
            create: {
              Codebase: {
                allowed: true,
              },
            },
            update: {},
            delete: {},
          },
        }}
      />
    </TestWrapper>
  );

  const dialog = screen.getByTestId('form');
  expect(dialog).toMatchSnapshot();
});
