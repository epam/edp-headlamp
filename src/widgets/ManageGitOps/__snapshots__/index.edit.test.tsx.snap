// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ManageGitOps Edit component 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  data-testid="form"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-jfdv4h-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Git server
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd  css-p2czvz-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-controls="mui-1"
                aria-disabled="true"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-gitServer"
                class="MuiSelect-select MuiSelect-outlined Mui-disabled MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-14oa7yt-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-gitServer"
                role="combobox"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                disabled=""
                name="gitServer"
                tabindex="-1"
                value="gerrit"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined Mui-disabled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <div
                class="MuiBox-root css-3avfg1"
              >
                <span />
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-jfdv4h-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="mui-4"
                id="mui-4-label"
              >
                Repository Name
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1i0uxaz-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-3mzles-MuiTypography-root"
                  >
                    /
                  </p>
                </div>
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id="mui-4"
                  name="name"
                  type="text"
                  value="edp-gitops"
                />
                <span />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      Repository Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1xu1mzw-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1llbb2a-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              undo changes
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-vw8gqz-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
